hivemind/dht/__init__.py

```diff
@@ -94,6 +94,9 @@ def __init__(
         self._client_mode = None
         self._p2p_replica = None
 
+        if "p2p" in self.kwargs:
+            self._daemon_listen_maddr = self.kwargs.pop("p2p").daemon_listen_maddr
```

Comment:
Let's add `p2p: P2P` to the `__init__` args and do it this way:

```suggestion
        self._daemon_listen_maddr = p2p.daemon_listen_maddr if p2p is not None else None
```

The difference is that (1) we won't keep the extra reference to the original P2P in `self.kwargs` (so the caller may destroy it if it wants to) and (2) we will have explicit argument instead of accessing kwargs (so we will still have type and name checking).